devicetypes/smartthings/cree-bulb.src/cree-bulb.groovy

```diff
@@ -82,7 +82,7 @@ def on() {
 }
 
 def setLevel(value) {
-    zigbee.setLevel(value) + ["delay 500"] + zigbee.levelRefresh()         //adding refresh because of ZLL bulb not conforming to send-me-a-report
+    zigbee.setLevel(value) + ["delay 500"] + zigbee.levelRefresh() + zigbee.onOffRefresh()       //adding refresh because of ZLL bulb not conforming to send-me-a-report
```

Comment:
Try removing the extra delay after `zigbee.setLevel` and putting the `zigbee.onOffRefresh` there instead. Then the on/off status will be updated more quickly and we'll still give the device time to transition to the new level before reading the current level again.